src/device/rttDevice.js

```diff
@@ -208,47 +178,46 @@ class RTTDevice extends Device {
 
     logProbeInfo() {
         return new Promise(resolve => {
-            nRFjprogjs.getProbeInfo(this.serialNumber, (err, info) => {
-                logger.info('SEGGER serial number: ', info.serialNumber);
-                logger.info('SEGGER speed: ', info.clockSpeedkHz, ' kHz');
-                logger.info('SEGGER version: ', info.firmwareString);
-                resolve();
-            });
+            logger.info('SEGGER serial number: ', this.device.serialNumber);
+            logger.info(
+                'SEGGER version: ',
+                this.device.jlink.jlinkObFirmwareVersion
+            );
+            resolve();
         });
     }
 
     startRTT() {
         return promiseTimeout(
             WAIT_FOR_START,
             new Promise((resolve, reject) => {
-                nRFjprogjs.rttStart(this.serialNumber, {}, err =>
-                    err ? reject(err) : resolve()
-                );
+                nRFDeviceLib
+                    .rttStart(deviceLibContext, this.device.id, 1000)
+                    .then(() => resolve())
+                    .catch(err => reject(err));
             })
         );
     }
 
-    static readRTT(serialNumber, length) {
+    static readRTT(deviceId, length) {
         return new Promise((resolve, reject) => {
-            nRFjprogjs.rttRead(
-                serialNumber,
-                0,
-                length,
-                (err, string, rawbytes, time) => {
-                    if (err) {
-                        return reject(new Error(err));
-                    }
-                    return resolve({ string, rawbytes, time });
-                }
-            );
+            nRFDeviceLib
+                .rttRead(deviceLibContext, deviceId, 0, length)
+                .then(({ rttReadBuffer: rawbytes }) => {
+                    return resolve({
+                        rawbytes,
+                    });
+                })
+                .catch(err => reject(new Error(err)));
```

Comment:
According to the type`rttReadBuffer` that is returned by the Promise from `nRFDeviceLib.rttRead()` is a Buffer, which usually does not contain a `rawbytes` property. I assume you did check, that this still works.

If so, then this can probably be simplified to 

```js
    static readRTT = async (deviceId, length) => {
        const { rttReadBuffer: rawbytes } = await nRFDeviceLib.rttRead(deviceLibContext, deviceId, 0, length)

        return { rawbytes }
    }
```